### ISSUE TITLE
Fix false positives for template literals

### ISSUE THREAD

**Issue by sathomas** [stylelint/stylelint#3218]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

When using a preprocessor such as `@mapbox/stylelint-processor-arbitrary-tags` no errors are reported, despite deliberate errors in the source files.

> Which rule, if any, is this issue related to?

all

> What CSS is needed to reproduce this issue?

The following `Test.vue` file:

```html
<docs></docs>
<template></template>
<script></script>
<style>
.selector {
  --property: value;;
}
</style>
```

> What stylelint configuration is needed to reproduce this issue?

The following `.stylelintrc`:

```json
{
  "extends": ["stylelint-config-standard", "stylelint-config-recommended-scss"],
  "processors": ["@mapbox/stylelint-processor-arbitrary-tags"],
  "rules": {
    "indentation": "tab",
    "no-empty-source": null
  }
}
```

> Which version of stylelint are you using?

`9.1.3`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint "Test.vue"`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

Same result as with version 9.1.2:

```
Test.vue
6:20  ✖  Unexpected extra semicolon   no-extra-semicolons
```

> What actually happened (e.g. what warnings or errors you are getting)?

No errors reported

**Comment by jeddy3** [comment]
@sathomas Thanks for the report and for using the template.

@gucong3000 Could the latest release of `postcss-html` have caused this regression?

**Comment by gucong3000** [comment]
After remove `processors` from `.stylelintrc`, it looks work fine.
```shell
E:\work\test  (test@0.0.0)
$ npm i stylelint stylelint-scss stylelint-config-standard stylelint-config-recommended-scss
npm WARN test@0.0.0 No description
npm WARN test@0.0.0 No repository field.
npm WARN test@0.0.0 No license field.

+ stylelint@9.1.3
+ stylelint-scss@2.5.0
+ stylelint-config-recommended-scss@3.1.0
+ stylelint-config-standard@18.2.0
added 3 packages, removed 20 packages, updated 22 packages and moved 15 packages in 61.424s

E:\work\test  (test@0.0.0)
$ stylelint "Test.vue"

Test.vue
 2:19  ×  Unexpected extra semicolon      no-extra-semicolons
 6:3   ×  Expected indentation of 1 tab   indentation

```

**Comment by jeddy3** [comment]
Oh, I see. Because stylelint can now, via the `postcss-html` syntax, parse Vue templates _out of the box_, the `stylelint-processor-arbitrary-tags` processor is now redundant. It also, for whatever reason, conflicts with it. 

How does this change affect processors like [`stylelint-processor-styled-components`](https://github.com/styled-components/stylelint-processor-styled-components)?

Is there anything we can do to remove the conflict so that using `stylelint-processor-arbitrary-tags` with Vue files isn't a problem? If not, how do we communicate that there is a conflict and the `stylelint-processor-arbitrary-tags` processor isn't needed for Vue files?

**Comment by sathomas** [comment]
> If not, how do we communicate that there is a conflict and the stylelint-processor-arbitrary-tags processor isn't needed for Vue files?

Note that because of #3219, removing the processor from the configuration still leaves stylelint v9.1.3 unusable.

**Comment by daKmoR** [comment]
seems to be I'm in the same boat?

my `.stylelintrc.js`
```
module.exports = {
  "extends": "stylelint-config-standard",
  "processors": [
    ["@mapbox/stylelint-processor-arbitrary-tags", { fileFilterRegex: [/\.html$/] }],
  ],
  "rules": {
    "no-empty-source": null,
  }
};
```

so if I read this correctly I no longer need `stylelint-processor-arbitrary-tags`?

but if I just remove it and do `stylelint file.html` I get nothing?

do I need to install `postcss-html` manually and configure it somehow?

**Comment by gucong3000** [comment]
@sathomas @daKmoR This is a demo here: [test.zip](https://github.com/stylelint/stylelint/files/1827568/test.zip)


```shell
E:\work\test  (test@0.0.0)
$ npm install-test
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN test@0.0.0 No description
npm WARN test@0.0.0 No repository field.
npm WARN test@0.0.0 No license field.

added 261 packages in 161.404s

> test@0.0.0 test E:\work\test
> stylelint Test.vue


Test.vue
 2:19  ×  Unexpected extra semicolon      no-extra-semicolons
```

**Comment by gucong3000** [comment]
> How does this change affect processors like stylelint-processor-styled-components?

https://github.com/stylelint/stylelint/blob/c3f4abebcbd869e5f14476e5c7829558ef0438c4/lib/getPostcssResult.js#L105-L109

Processors will just remove non-CSS blocks from code, we should keep non-CSS blocks.

**Comment by daKmoR** [comment]
@gucong3000 
I made a demo for js with template literals (where it works) and for the same code in a
 `script` tag in html where it does not work.
[test-js-html.zip](https://github.com/stylelint/stylelint/files/1828268/test-js-html.zip)

`Test.html` (won't show errors)
```html
<script>
  function test() {
    return `
      <style>
        .selector {
          --property: value;;
        }
      </style>
    `;
  }
</script>
```

`Test.js` (shows errors)
```js
function test() {
  return `
    <style>
      .selector {
        --property: value;;
      }
    </style>
  `;
}
```

Log:
```
$ npm run test:js

> test@0.0.0 test:js /vagrant/new/test-js-html
> stylelint Test.js


Test.js
 3:10  ✖  Unexpected extra semicolon   no-extra-semicolons

$ npm run test:html

> test@0.0.0 test:html /vagrant/new/test-js-html
> stylelint Test.html
```

I guess I will need to do something as it's probably not the default use case... but I'm not sure what as with processors it won't work as well...

**Comment by gucong3000** [comment]
@daKmoR I opened a new issues for this problem.
https://github.com/gucong3000/postcss-html/issues/32

**Comment by jeddy3** [comment]
Done in https://github.com/stylelint/stylelint/pull/3238
